connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -403,7 +403,7 @@ private void convertMessages(ConsumerRecords<byte[], byte[]> msgs) {
                     keyAndSchema.schema(), keyAndSchema.value(),
                     valueAndSchema.schema(), valueAndSchema.value(),
                     msg.offset(),
-                    (msg.timestampType() == TimestampType.NO_TIMESTAMP_TYPE) ? null : msg.timestamp(),
+                    (msg.timestamp() <= Record.NO_TIMESTAMP) ? null : msg.timestamp(),
```

Comment:
This is a bit weird since it's basically assuming you know that `NO_TIMESTAMP` is -1.